client/my-sites/hosting/web-server-logs-card/index.js

```diff
@@ -18,6 +20,192 @@ import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selecto
+			if ( startOrEnd === 'Start' ) {
+				validationResults = onValidate( newValue, endValue );
+			}
+			if ( startOrEnd === 'End' ) {
+				validationResults = onValidate( startValue, newValue );
+			}
+			if ( ! validationResults || ! Array.isArray( validationResults ) ) {
+				return null;
+			}
+			return validationResults;
+		},
+		[ endValue, startValue, onValidate ]
+	);
+
+	/**
+	 * Handles input focus events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  startOrEnd one of "Start" or "End"
+	 * @returns the partially applied function ready to receive event data
+	 */
+	const handleInputFocus = useCallback(
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( ! showValidation ) {
+				return onInputFocus( value, startOrEnd );
+			}
+			const validationResults = validate( startOrEnd, value );
+			if ( ! validationResults ) {
+				return onInputFocus( value, startOrEnd );
+			}
+			const [ startValidatorResults, endValidatorResults ] = validationResults;
+			if (
+				( startOrEnd === 'Start' && startValidatorResults.isValid ) ||
+				( startOrEnd === 'End' && endValidatorResults.isValid )
+			) {
+				return onInputFocus( value, startOrEnd );
+			}
+		},
+		[ onInputFocus, validate, showValidation ]
+	);
+
+	/**
+	 * Handles input blur events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  startOrEnd one of "Start" or "End"
+	 * @returns the partially applied function ready to receive event data
+	 */
+	const handleInputBlur = useCallback(
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( ! showValidation ) {
+				return onInputBlur( value, startOrEnd );
+			}
+			const validationResults = validate( startOrEnd, value );
+			if ( ! validationResults || validationResults.isValid ) {
+				onInputBlur( value, startOrEnd );
+			}
+
+			if ( ! validationResults ) {
+				return onInputBlur( value, startOrEnd );
+			}
+			const [ startValidatorResults, endValidatorResults ] = validationResults;
+			if (
+				( startOrEnd === 'Start' && startValidatorResults.isValid ) ||
+				( startOrEnd === 'End' && endValidatorResults.isValid )
+			) {
+				return onInputBlur( value, startOrEnd );
+			}
+		},
+		[ onInputBlur, validate, showValidation ]
+	);
+
+	/**
+	 * Handles input change events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  startOrEnd one of "Start" or "End"
+	 * @returns the partially applied function ready to receive event data
+	 */
+	const handleInputChange = useCallback(
+		( startOrEnd ) => ( e ) => {
+			const { value } = e.target;
+			if ( showValidation ) {
+				validate( startOrEnd, value );
+			}
+			onInputChange( value, startOrEnd );
+		},
+		[ onInputChange, validate, showValidation ]
+	);
+
+	return (
+		<FormFieldset className="date-range__date-inputs">
+			<legend className="date-range__date-inputs-legend">
+				{ translate( 'Start and End Dates' ) }
```

Comment:
```suggestion
				{ translate( 'Start and end dates' ) }
```

Labels should be sentence-case, see conversation in p1676381941241629-slack-C9EJ7KSGH

Reply:
Should I update this component as well?
https://github.com/Automattic/wp-calypso/blob/trunk/client/components/date-range/inputs.tsx#L94

Reply:
@kozer Probably outside the scope of this PR. You could do so in a separate PR, though.